diff --git a/aconfigure b/aconfigure
index 784838c..9533f55 100755
--- a/aconfigure
+++ b/aconfigure
@@ -644,6 +644,7 @@ ac_resample_dll
 ac_no_ilbc_codec
 ac_no_speex_codec
 ac_no_g7221_codec
+ac_no_g729_codec
 ac_no_g722_codec
 ac_no_gsm_codec
 ac_no_l16_codec
@@ -762,6 +763,7 @@ enable_l16_codec
 enable_gsm_codec
 enable_g722_codec
 enable_g7221_codec
+enable_g729_codec
 enable_speex_codec
 enable_ilbc_codec
 enable_libsamplerate
@@ -1421,6 +1423,7 @@ Optional Features:
   --disable-gsm-codec     Exclude GSM codec in the build
   --disable-g722-codec    Exclude G.722 codec in the build
   --disable-g7221-codec   Exclude G.7221 codec in the build
+  --disable-g729-codec    Exclude G.729 codec in the build
   --disable-speex-codec   Exclude Speex codecs in the build
   --disable-ilbc-codec    Exclude iLBC codec in the build
   --enable-libsamplerate  Link with libsamplerate when available. Note that
@@ -5948,6 +5951,20 @@ $as_echo "Checking if G.722.1 codec is disabled...no" >&6; }
 fi
 
 
+# Check whether --enable-g7221-codec was given.
+if test "${enable_g729_codec+set}" = set; then :
+  enableval=$enable_g729_codec; if test "$enable_g729_codec" = "no"; then
+		ac_no_g729_codec=1
+		$as_echo "#define PJMEDIA_HAS_G729_CODEC 0" >>confdefs.h
+
+		{ $as_echo "$as_me:${as_lineno-$LINENO}: result: Checking if G.729 codec is disabled...yes" >&5
+$as_echo "Checking if G.729 codec is disabled...yes" >&6; }
+	       fi
+else
+  { $as_echo "$as_me:${as_lineno-$LINENO}: result: Checking if G.729 codec is disabled...no" >&5
+$as_echo "Checking if G.729 codec is disabled...no" >&6; }
+fi
+
 
 # Check whether --enable-speex-codec was given.
 if test "${enable_speex_codec+set}" = set; then :
diff --git a/aconfigure.ac b/aconfigure.ac
index be2f2ae..ae47ea1 100644
--- a/aconfigure.ac
+++ b/aconfigure.ac
@@ -785,6 +785,18 @@ AC_ARG_ENABLE(g7221-codec,
 	       fi],
 	      AC_MSG_RESULT([Checking if G.722.1 codec is disabled...no]))
 
+dnl # Include G729 codec
+AC_SUBST(ac_no_g729_codec)
+AC_ARG_ENABLE(g729-codec,
+	      AC_HELP_STRING([--disable-g729-codec],
+			     [Exclude G.729 codec in the build]),
+	      [if test "$enable_g729_codec" = "no"; then
+		[ac_no_g729_codec=1]
+		AC_DEFINE(PJMEDIA_HAS_G729_CODEC,0)
+		AC_MSG_RESULT([Checking if G.729 codec is disabled...yes])
+	       fi],
+	      AC_MSG_RESULT([Checking if G.729 codec is disabled...no]))
+
 dnl # Include Speex codec
 AC_SUBST(ac_no_speex_codec)
 AC_ARG_ENABLE(speex-codec,
diff --git a/build.mak.in b/build.mak.in
index 5ea6e79..2c8b9df 100644
--- a/build.mak.in
+++ b/build.mak.in
@@ -59,6 +59,11 @@ APP_THIRD_PARTY_LIBS += -lg7221codec-$(TARGET_NAME)
 APP_THIRD_PARTY_LIB_FILES += $(PJ_DIR)/third_party/lib/libg7221codec-$(LIB_SUFFIX)
 endif
 
+ifneq (@ac_no_g729_codec@,1)
+APP_THIRD_PARTY_LIBS += -lg729codec-$(TARGET_NAME)
+APP_THIRD_PARTY_LIB_FILES += $(PJ_DIR)/third_party/lib/libg729codec-$(LIB_SUFFIX)
+endif
+
 ifneq ($(findstring pa,@ac_pjmedia_snd@),)
 ifeq (@ac_external_pa@,1)
 # External PA
diff --git a/pjmedia/build/Makefile b/pjmedia/build/Makefile
index 2f97296..70a34a0 100644
--- a/pjmedia/build/Makefile
+++ b/pjmedia/build/Makefile
@@ -114,9 +114,9 @@ export PJMEDIA_CODEC_OBJS += audio_codecs.o ffmpeg_vid_codecs.o \
 			h263_packetizer.o h264_packetizer.o \
 			$(OS_OBJS) $(M_OBJS) $(CC_OBJS) $(HOST_OBJS) \
 			ipp_codecs.o $(CODEC_OBJS) \
-                        g7221_sdp_match.o amr_sdp_match.o
+                        g7221_sdp_match.o amr_sdp_match.o g729.o
 export PJMEDIA_CODEC_CFLAGS += $(_CFLAGS) $(GSM_CFLAGS) $(SPEEX_CFLAGS) \
-			$(ILBC_CFLAGS) $(IPP_CFLAGS) $(G7221_CFLAGS)
+			$(ILBC_CFLAGS) $(IPP_CFLAGS) $(G7221_CFLAGS) $(G729_CFLAGS)
 
 
 ###############################################################################
diff --git a/pjmedia/build/os-darwinos.mak b/pjmedia/build/os-darwinos.mak
index 7904594..020e67d 100644
--- a/pjmedia/build/os-darwinos.mak
+++ b/pjmedia/build/os-darwinos.mak
@@ -25,6 +25,7 @@ AC_NO_SPEEX_CODEC=0
 AC_NO_ILBC_CODEC=0
 AC_NO_G722_CODEC=0
 AC_NO_G7221_CODEC=0
+AC_NO_G729_CODEC=0
 
 export CODEC_OBJS=
 
diff --git a/pjmedia/include/pjmedia-codec/config.h b/pjmedia/include/pjmedia-codec/config.h
index 99b1d91..fa52475 100644
--- a/pjmedia/include/pjmedia-codec/config.h
+++ b/pjmedia/include/pjmedia-codec/config.h
@@ -313,6 +313,13 @@
 #endif
 
 /**
+ * G729 codec is disabled by default
+ */
+#ifndef PJMEDIA_HAS_G729_CODEC
+#   define PJMEDIA_HAS_G729_CODEC 0
+#endif
+
+/**
  * Enable OpenCORE AMR-NB codec.
  * See https://trac.pjsip.org/repos/ticket/1388 for some info.
  *
diff --git a/pjmedia/include/pjmedia-codec/config_auto.h.in b/pjmedia/include/pjmedia-codec/config_auto.h.in
index c469d79..56afd1b 100644
--- a/pjmedia/include/pjmedia-codec/config_auto.h.in
+++ b/pjmedia/include/pjmedia-codec/config_auto.h.in
@@ -69,6 +69,11 @@
 #undef PJMEDIA_HAS_G7221_CODEC
 #endif
 
+/* G729 codec */
+#ifndef PJMEDIA_HAS_G729_CODEC
+#undef PJMEDIA_HAS_G729_CODEC
+#endif
+
 /* OpenCORE AMR-NB codec */
 #ifndef PJMEDIA_HAS_OPENCORE_AMRNB_CODEC
 #undef PJMEDIA_HAS_OPENCORE_AMRNB_CODEC
diff --git a/pjmedia/src/pjmedia-codec/audio_codecs.c b/pjmedia/src/pjmedia-codec/audio_codecs.c
index 1fbd038..1f6110f 100644
--- a/pjmedia/src/pjmedia-codec/audio_codecs.c
+++ b/pjmedia/src/pjmedia-codec/audio_codecs.c
@@ -100,6 +100,12 @@ pjmedia_codec_register_audio_codecs(pjmedia_endpt *endpt,
 	return status;
 #endif /* PJMEDIA_HAS_G7221_CODEC */
 
+#if PJMEDIA_HAS_G729_CODEC
+    status = pjmedia_codec_g729_init(endpt);
+    if (status != PJ_SUCCESS)
+      return status;
+#endif /* PJMEDIA_HAS_G729_CODEC */
+
 #if PJMEDIA_HAS_L16_CODEC
     /* Register L16 family codecs */
     status = pjmedia_codec_l16_init(endpt, 0);
